### DISCUSSION TITLE
Adding a Tile option for the watch OS version

### DISCUSSION THREAD

**Discussion by rdpeake** [NightscoutFoundation/xDrip#3138] [Ideas]
The graph display for the xdrip watch face is useful, but not really something i feel i need on the watch at all times. Is it possible to have the more advance display as a watch Tile service and then the main watch face can simply use the xdrip customize source?

If not, is there a way to gain access to the xdrip data as a 3rd party app on the watch to then make such a tile as a separate app?

**Comment by corneliusroemer** [comment]
I think you can get xdrip to broadcast data to other apps on the phone to make use of. So in theory it should be possible to write a tile app that does what you want. I have thought of doing something like that because xdrip doesn't seem to work with wear os 4 with android 14.

**Reply by DrMacinyasha** [reply]
FWIW, I was able to get xDrip+'s watchface working with Wear OS 4 and Android 14 (Pixel 8 Pro, Pixel Watch 2). You have to go through a bit of a complicated process to sideload the watch app using Wear Installer 2 on your phone, and connect the devices on the same Wi-Fi network, enable ADB and wireless debugging on the watch, pair the app to the watch, and *then* you can install the watchface. 

However, I 100% agree with @rdpeake; a tile for xDrip+, just like Fitbit or Google Assistant, would be greatly appreciated.

**Comment by mprosenberg** [comment]
I have designed a couple of watch faces that work on Tizen based watches, and should work on Wear Os based watches. They are available on Watchmaker - search for mprosenberg, or xdrip+. Tapping the bottom of the watch face opens a second screen showing the graph, high, low, avg, and change/update. Here are some screen shots. I can up load other watch faces with the same feature. The first image is the watch face, and the second image is the second screen; you can tap the center of the second screen to bring up the 4 hr, and 6 hr screen. In addition, the color of watch face components change color with blood sugar.  
![Screenshot_20231210_110812_WatchMaker](https://github.com/NightscoutFoundation/xDrip/assets/58182423/ad6a342b-68c9-4003-9bcd-b22cc52c5e4b)    
![Screenshot_20231210_110820_WatchMaker](https://github.com/NightscoutFoundation/xDrip/assets/58182423/fdfcea25-adcd-46ad-825c-062499646520)

Mike

**Comment by Ukcraig** [comment]
I came here looking for an answer to this exact question. Having the data on a complication is good but someone I do want to see the full graph and it would be great to have it on a tile.

**Comment by Navid200** [comment]
Unfortunately, no developer is providing support for WearOS right now.  This has been the case for quite a while may be a year if not more.

Everyone is welcome to contribute considering this is an open source project.  

MiBand and may be more are supported by a companion app called WatchDrip.
Ideally, a developer will create a similar companion app for WearOS.

**Comment by tjggriffin** [comment]
I use Wearable Widgets to send the xDrip phone widget to my Galaxy Watch 4. Then I put WW on my watch as the 2nd tile.
